resources/js/views/rooms/View.vue

```diff
@@ -282,71 +282,43 @@ export default {
     setInterval(this.reload, env.REFRESH_RATE * 1000);
   },
   methods: {
-
     /**
-     * Request file download url
-     * @param file file object
-     * @param index integer index in filelist
-     * @return string url
+     *  Handle errors of the file list
      */
-    downloadFile: function (file, index) {
-      this.loadingDownload = true;
-      // Update value for the setting and the effected file
-      const config = this.accessCode == null ? {} : { headers: { 'Access-Code': this.accessCode } };
-      Base.call('rooms/' + this.room.id + '/files/' + file.id, config)
-        .then(response => {
-          if (response.data.url !== undefined) {
-            window.open(response.data.url, '_blank');
-          }
-        }).catch((error) => {
-          if (error.response) {
-          // Access code invalid
-            if (error.response.status === 401 && error.response.data.message === 'invalid_code') {
-            // Show access code is valid
-              this.accessCodeValid = false;
-              // Reset access code (not the form input) to load the general room details again
-              this.accessCode = null;
-              // Show error message
-              this.flashMessage.error(this.$t('rooms.flash.accessCodeChanged'));
-              this.reload();
-              return;
-            }
+    onFileListError: function (error) {
+      if (error.response) {
+        // Access code invalid
+        if (error.response.status === 401 && error.response.data.message === 'invalid_code') {
+          return this.handleInvalidCode();
+        }
 
-            // Forbidden, require access code
-            if (error.response.status === 403 && error.response.data.message === 'require_code') {
-            // Show access code is valid
-              this.accessCodeValid = false;
-              // Reset access code (not the form input) to load the general room details again
-              this.accessCode = null;
-              // Show error message
-              this.flashMessage.error(this.$t('rooms.flash.accessCodeChanged'));
-              this.reload();
-              return;
-            }
+        // Access code invalid
+        if (error.response.status === 403 && error.response.data.message === 'require_code') {
+          return this.handleInvalidCode();
+        }
 
-            // Forbidden, not allowed to download this file
-            if (error.response.status === 403) {
-            // Show error message
-              this.flashMessage.error(this.$t('rooms.flash.fileForbidden'));
-              // Remove file from list
-              this.room.files.splice(index, 1);
-              return;
-            }
+        // Forbidden, guests not allowed
+        if (error.response.status === 403) {
+          this.room = null;
```

Comment:
This code already exists in line 356 and following. Please define also a function for this case.